src/html5_parser/parser/document.rs

```diff
@@ -32,23 +37,97 @@ impl Document {
         arena.add_node(Node::new_document());
         Self {
             arena,
+            named_id_elements: HashMap::new(),
             doctype: DocumentType::HTML,
             quirks_mode: QuirksMode::NoQuirks,
         }
     }
 
-    // Fetches a node by id or returns None when no node with this ID is found
+    /// Fetches a node by id or returns None when no node with this ID is found
     pub fn get_node_by_id(&self, node_id: NodeId) -> Option<&Node> {
         self.arena.get_node(node_id)
     }
 
+    /// Fetches a mutable node by id or returns None when no node with this ID is found
     pub fn get_mut_node_by_id(&mut self, node_id: NodeId) -> Option<&mut Node> {
         self.arena.get_mut_node(node_id)
     }
 
+    /// Fetches a node by named id (string) or returns None when no node with this ID is found
+    pub fn get_node_by_named_id(&self, named_id: &str) -> Option<&Node> {
+        let node_id = self.named_id_elements.get(named_id)?;
+        self.arena.get_node(*node_id)
+    }
+
+    /// Fetches a mutable node by named id (string) or returns None when no node with this ID is found
+    pub fn get_mut_node_by_named_id(&mut self, named_id: &str) -> Option<&mut Node> {
```

Comment:
How about `_mut` as the final suffix instead of being in the middle of the function name, here and elsewhere?

Reply:
yes, i'll adjust these